### ISSUE TITLE
zathura and syntex

### ISSUE THREAD

**Issue by coachshea** [lervag/vimtex#1083]
# Explain the issue

When opening zathura with <plug>(vimtex-view), syntex forward search works as expected (i.e. zathura opens, has the focus, and it is opened to the correct location), but when I leave zathura (without closing it) and go back to my project and use <plug>(vimtex-view) again. Nothing happens. If I manually switch focus to the zathura instance, I see that forward search has worked, but zathura didn't get the focus, until I manualy switched it. If I close zathura and execute <plug>(vimtex-view), then it works correctly. 

For what it's worth, Backward search works consistently. Neovim gets the focus, the correct file is opened (if it wasn't already) and focused, and it is scrolled to the correct line.

1. A minimal `init.vim` file, for instance:

```vim
call plug#begin()
Plug 'lervag/vimtex'
call plug#end()

let g:vimtex_enabled=1
let g:vimtex_quickfix_mode=2
let g:vimtex_fold_enabled=1
let g:vimtex_motion_enabled=1
let g:vimtex_mappings_enabled=0
let g:vimtex_imaps_enabled=0
let g:vimtex_latexmk_options='-bibtex -pdf'
let g:vimtex_latexmk_build_dir='./build'
let g:vimtex_complete_close_braces=1
let g:vimtex_complete_recursive_bib=1
let g:vimtex_view_general_viewer='zathura'
let g:vimtex_view_method='zathura'
let g:vimtex_compiler_progname = 'nvr'
```

2. A minimal `.tex` file, such as:

    ```tex
    \documentclass{minimal}
    \begin{document}

    Any large multi-file tex project will do.

    \end{document}
    ```

3. A concise description of the issue and **the steps to reproduce** it. This should include something like `vim --servername VIM -u vimrc mwe.tex` to show that one understands the use of minimal vimrc files.

I execute <plug>(vimtex-view) and nothing appears to happen (although the document is forward search). I 
 expect zathura to get the focus.

4. A `.latexmkrc` file [*if relevant*].

$pdflatex = 'pdflatex --shell-escape -synctex=1 %O %S';

$out_dir = './build'

**Comment by coachshea** [comment]
Sorry, I was looking at this and realized how ugly it came out. I was copying and pasting and I must've accidentally messed up the formatting. I should've checked the preview before posting. If you'd like, I can close this issue and open a new one that is more readable.

**Comment by andymass** [comment]
@coachshea you can edit your posts on github.  And although this is not exactly a reproducible MWE, I think I can reproduce this overall issue in the sense that vimtex has never "re-focused" zathura or brought it to the top when issuing `<plug>(vimtex-view)`.

**Comment by coachshea** [comment]
Thank you for your reply,  I have edited the original comment. I didn't realize that's how it's supposed to work. I'll close this issue.

**Comment by lervag** [comment]
First, I'll allow myself to comment on the issue description: There was a lot of boiler plate issue content left over, and there were unnecessary details. I've edited the original post to simplify to the required level of detail. I understand that it might be difficult to know exactly which details are necessary, so the only thing I'll comment on here is that you should make your own wording; the issue description that's there as a boiler plate issue is only to suggest which content is the most relevant to include.

So, to the issue at hand: it is possible to change this behaviour to work like you would expect. But it is not clear to me that this is the desired behaviour. I'm often doing `\lv` to update the PDF viewer position, but I still want to work on the document in Vim. Which actions would you want to do in the viewer anyway? If you do `\lv` for a forward search, that should already be sort of "sufficient" for a motion in the viewer...

**Comment by andymass** [comment]
I agree, I wouldn't want this to be the default.  I personally lose track of my window occasionally and like to have a way to bring it to the top.  Hence, I've added this to my `ftplugin/tex.vim`:

```vim
function! s:FocusTexViewer()
  if b:vimtex.viewer.xwin_id
    call system('xdotool windowactivate '.b:vimtex.viewer.xwin_id.' &')
  endif
  VimtexView
endfunction
nnoremap <silent><buffer> <leader>lV :<c-u>call <sid>FocusTexViewer()<cr>
```

**Comment by coachshea** [comment]
First, @lervag  I apologize for the clumsiness of my original post. It's been a long time since I've posted an issue here (although I use vimtex frequently and am a huge fan). I don't believe that the template existed the last time that I posted (or if so I don't remember). I did my best to make my question fit into the template, I'm sorry for any confusion that my poorly organized post caused.

To the issue, I certainly don't expect a feature to be added for one person and I am going to give the suggestion by @andymass a chance, it looks like a great idea, Thank you for posting it. However, I will try to explain my use case for whatever it might be worth.

I tend to work my windows in full screen mode. I feel as though I am least distracted and most productive that way. So, I am usually working with vim in full screen mode. When I've done enough work that I want to inspect the pdf output, I type <leader>lv and Zathura pops open in full screen mode and to the correct location. I review what I need to and then I go back to vim in full screen mode. I had previously quit Zathura after viewing it. Zathura loads so fast that I never gave it a thought. Lately, I've been using backwards search and when I CTRL-click, vim pops open in full screen mode again and I get back to work (but now Zathura stays open. To my surprise, when I now type <leader>lv, nothing appears to happen. Of course, forward search is happening, but I can't see it. Having said all that, I understand that not everyone works that way and I will give the above solution a try. As I said, it looks like a great fit for what I need.

Thanks to both of you for all your help.

**Comment by lervag** [comment]
Another possibility is to remap `\lv` like this:

```vim
nnoremap <silent><buffer> <leader>lv :<c-u>call <sid>OpenViewer()<cr>

function! s:OpenViewer()
  " Open the viewer and/or do forward search
  VimtexView

  " Add a very small amount of sleep time to allow previous process to finish,
  " you should experiment with the exact amount. Perhaps not necessary at all.
  sleep 50m

  " Now use xdotool to focus the viewer
  if b:vimtex.viewer.xwin_id
    call system('xdotool windowactivate '.b:vimtex.viewer.xwin_id.' &')
  endif
endfunction
```

This adds a wrapper around the original mapping (to `:VimtexView`) that keeps the original behaviour and appends your desired focussing behaviour.

**Comment by coachshea** [comment]
Thanks again for all the suggestions. After playing around with these great suggestions, I decided to do do as suggested. I have remapped <leader>lv and it is working great. A couple of points that I will include in case someone else comes across this:

First, I didn't need 'sleep' and it didn't matter if 'VimtexView' was place before or after the conditional 'xdotool' call. In both cases, within a second, I was viewing the pdf and it was forward searched.

Second, I had to use 'xdotool windowfocus ', for whatever reason 'windowactivate' didn't work for me.

Lastly, I didn't need to include the trailing ' &'. I've never used xdotool much before, but after playing around with it for a while, it appears that this feature is built into xdotool. It seems to automatically return from it's command and doesn't require the '&'.

Thank again @andymass and @lervag. Your help was greatly appreciated. I never would have come up with this on my own.
